conda_build/post.py

```diff
@@ -70,10 +70,13 @@ def fix_shebang(f, osx_is_app=False):
     py_exec = ('/bin/bash ' + config.build_prefix + '/bin/python.app'
                if sys.platform == 'darwin' and osx_is_app else
                config.build_prefix + '/bin/' + basename(config.build_python))
-    new_data = SHEBANG_PAT.sub(b'#!' + py_exec.encode(encoding), data, count=1)
+    lib = "{0}_LIBRARY_PATH={1}/lib".format('DYLD' if sys.platform == 'darwin' else 'LD',
+                                            config.build_prefix)
+    new_data = SHEBANG_PAT.sub('#!/usr/bin/env {0} '.format(lib) + py_exec.encode(encoding),
+                               data, count=1)
```

Comment:
@stuarteberg can you expand on "usually best"?  I think `DYLD_LIBRARY_PATH` is the appropriate environment variable in this particular case.